### PR TITLE
Add model_id_attr to phase change level CLI and related CLIs

### PR DESCRIPTION
**Pull request by gavinevans**
Related to https://github.com/metoppv/mo-blue-team/issues/354

Description
Add the possible provision of model_id_attr to the phase change level CLI and precursor wet bulb temperature and wet bulb temperature integral CLIs. This is with the aim of allowing a forthcoming hail fraction CLI to take a model_id_attr argument, which will take the output of the phase change level CLI as input. 

Please see [this comment](https://github.com/metoppv/mo-blue-team/issues/354#issuecomment-1249563395) for acceptance test data information.

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

File: improver_tests/acceptance/SHA256SUMS [hunk 1]
```diff
@@ -460,14 +460,20 @@ df2a6ed228f820901aa00ff0ac2b6e68ae14f0751b2d887fbde7a657845b2428  ./nowcast-opti
...
+6b7b0ea690520e1f7ebfd7c8e7cc7c645964010f98cbf49d472d49fdb4ef3a4b  ./phase-change-level/with_id_attr/snow_sleet_kgo.nc
+651c80cd4ab7c720c59c6c0c2c4a2ade85aee81c2ab0c697b577ca87a36d57f2  ./phase-change-level/with_id_attr/wbti.nc
+c2b71855f260544ca75aaa5900f3297b58cee1b3b64eec5805662e1927d5fdab  ./phase-change-level/with_id_attr/wet_bulb_temperature.nc
+8e73bf41e6d275f80ef27cb19491f70b55cd72551f504f4ba9b8a86bc619bd6f  ./phase-change-level/without_id_attr/hail_rain_kgo.nc
+7beb5a535688252eb81222d2bfa13a82306d4c3da3990b1ed09f850431495fbb  ./phase-change-level/without_id_attr/sleet_rain_kgo.nc
+a87b4eb0ae9dfac3cdffb75dba94750e047fe618b4bb469f5e2262819bfd2318  ./phase-change-level/without_id_attr/sleet_rain_unfilled_kgo.nc
+ce8c714663974d489b25516f752c21424c52182dacc930f78167c6fbfe7e3ac2  ./phase-change-level/without_id_attr/snow_sleet_kgo.nc
+8693b7061a6a3048b193b084e23f9c782a655426764a2c3d1b77ea456d883c09  ./phase-change-level/without_id_attr/wbti.nc
+95c8e9238f70628bfa19276cf32c401460fee518411a51286c25de3598656e85  ./phase-change-level/without_id_attr/wet_bulb_temperature.nc
 d6f4cc94f3e317b9cad58f5f2a7f4147532b21d9bddd89df79158b44a18ee95e  ./phase-mask/rain/kgo.nc
```
Comment (MoseleyS) [inline improver_tests/acceptance/SHA256SUMS:477]: We could save space by just having one version of the inputs (`with_id_attr`) as the id attribute will be ignored by the `without` test.

File: improver_tests/acceptance/SHA256SUMS [hunk 2]
```diff
@@ -460,14 +460,20 @@ df2a6ed228f820901aa00ff0ac2b6e68ae14f0751b2d887fbde7a657845b2428  ./nowcast-opti
...
-8e73bf41e6d275f80ef27cb19491f70b55cd72551f504f4ba9b8a86bc619bd6f  ./phase-change-level/basic/hail_rain_kgo.nc
-16c0578e606553d4e041cf75e3bfbdbfdb78a0dc02f1f23099b580307951a9d0  ./phase-change-level/basic/land_mask.nc
-9623c8f2096b440b69b7e0d448a457ec70b10f88df6669cd78572642673e74a7  ./phase-change-level/basic/orog.nc
-7beb5a535688252eb81222d2bfa13a82306d4c3da3990b1ed09f850431495fbb  ./phase-change-level/basic/sleet_rain_kgo.nc
-a87b4eb0ae9dfac3cdffb75dba94750e047fe618b4bb469f5e2262819bfd2318  ./phase-change-level/basic/sleet_rain_unfilled_kgo.nc
-ce8c714663974d489b25516f752c21424c52182dacc930f78167c6fbfe7e3ac2  ./phase-change-level/basic/snow_sleet_kgo.nc
-8693b7061a6a3048b193b084e23f9c782a655426764a2c3d1b77ea456d883c09  ./phase-change-level/basic/wbti.nc
-95c8e9238f70628bfa19276cf32c401460fee518411a51286c25de3598656e85  ./phase-change-level/basic/wet_bulb_temperature.nc
+16c0578e606553d4e041cf75e3bfbdbfdb78a0dc02f1f23099b580307951a9d0  ./phase-change-level/land_mask.nc
+9623c8f2096b440b69b7e0d448a457ec70b10f88df6669cd78572642673e74a7  ./phase-change-level/orog.nc
```
Comment (MoseleyS) [inline improver_tests/acceptance/SHA256SUMS:464]: This file is also used by `test_phase_probability.nc`, which needs updating to be able to find it.

File: improver_tests/acceptance/SHA256SUMS [hunk 3]
```diff
@@ -658,14 +664,20 @@ d197c8f5f1a08caddaa5efabac4a52d75185889d64d2c4b72cbba2f2c0f42a20  ./weighted_ble
 30062d9c23527634a20cae6e7cf7a163aaee361b06ee800c8bea70a7a0f412ec  ./weighted_blending/three_models/nc/20190101T0400Z-PT0001H00M-precip_rate.nc
 850d48dfea1d1e5b696324f0e57a52fce3e3128008c58ccaca82e31598196c67  ./weighted_blending/three_models/ukvx/20190101T0400Z-PT0002H00M-precip_rate.nc
 63b4691e4d7e246b58e01c84f3c8cf82a6cdc7e14068ddd96418e8c0054e2bcf  ./weighted_blending/weights_from_dict/kgo.nc
-124a429618807548ee8890a2e1d4275be0e1e7977d9cfc804e55892bb045d323  ./wet-bulb-temperature-integral/basic/input.nc
-e3bef1754689e3ed8760774bfd293f0fee8ee647c76b308cf7a5a75fac22369e  ./wet-bulb-temperature-integral/basic/kgo.nc
+c2b71855f260544ca75aaa5900f3297b58cee1b3b64eec5805662e1927d5fdab  ./wet-bulb-temperature-integral/basic/with_id_attr/input.nc
+651c80cd4ab7c720c59c6c0c2c4a2ade85aee81c2ab0c697b577ca87a36d57f2  ./wet-bulb-temperature-integral/basic/with_id_attr/kgo.nc
+124a429618807548ee8890a2e1d4275be0e1e7977d9cfc804e55892bb045d323  ./wet-bulb-temperature-integral/basic/without_id_attr/input.nc
```
Comment (MoseleyS) [inline improver_tests/acceptance/SHA256SUMS:669]: Again, I don't think we need two versions of the input

File: improver_tests/acceptance/SHA256SUMS [hunk 4]
```diff
@@ -658,14 +664,20 @@ d197c8f5f1a08caddaa5efabac4a52d75185889d64d2c4b72cbba2f2c0f42a20  ./weighted_ble
...
-81d3480537dd30284a9745e152371d2ce529818a5e82fa26cb4d41a5b3969f3b  ./wet-bulb-temperature/basic/enukx_temperature.nc
-f3b782a8f7bfeb1ce0c5ffbca50b5cc41b02ff1637a1a5e37ec221daf893e36a  ./wet-bulb-temperature/basic/kgo.nc
+52de37aa15c072e30817327e40e8d9f0e60924d9164f6458e5633072ae2e04ed  ./wet-bulb-temperature/basic/with_id_attr/enukx_pressure.nc
+9e90b526a2355df27d3d29f5548e44d4ff8cae60bea0b9da071e5635a12d7c49  ./wet-bulb-temperature/basic/with_id_attr/enukx_relative_humidity.nc
+81d3480537dd30284a9745e152371d2ce529818a5e82fa26cb4d41a5b3969f3b  ./wet-bulb-temperature/basic/with_id_attr/enukx_temperature.nc
+7facf1f9031b06bb9384fb2a5c09efb134c95ddee8426d800d8ec92f4acdc11c  ./wet-bulb-temperature/basic/with_id_attr/kgo.nc
+5a849a09ec437b8dc824e9c8ab460957ebb5b98c199aa729c60bb89c018ffe20  ./wet-bulb-temperature/basic/without_id_attr/enukx_pressure.nc
+c1251bc099e5ac8d2acd4b7a05475766e646c33aeea3e15f8c679a59eb73b6eb  ./wet-bulb-temperature/basic/without_id_attr/enukx_relative_humidity.nc
+bdd00fbe95b4e9b4a1f9c758e21fda4a97f4eadca02174df695a824d4fd85528  ./wet-bulb-temperature/basic/without_id_attr/enukx_temperature.nc
```
Comment (MoseleyS) [inline improver_tests/acceptance/SHA256SUMS:679]: Same with these inputs

**Review by MoseleyS — CHANGES_REQUESTED** [review]
There are some failing tests to fix and also some duplicate data files that I think should be singular.

File: improver_tests/acceptance/SHA256SUMS (hunk 1 above)
Comment (gavinevans) [reply improver_tests/acceptance/SHA256SUMS]: I've reduced the duplication in the test data as suggested.

File: improver_tests/acceptance/SHA256SUMS (hunk 2 above)
Comment (gavinevans) [reply improver_tests/acceptance/SHA256SUMS:464]: Updated.

File: improver_tests/acceptance/SHA256SUMS (hunk 3 above)
Comment (gavinevans) [reply improver_tests/acceptance/SHA256SUMS]: Duplication removed.

File: improver_tests/acceptance/SHA256SUMS (hunk 4 above)
Comment (gavinevans) [reply improver_tests/acceptance/SHA256SUMS]: Duplication removed.

**Review by MoseleyS — DISMISSED** [review]

**Review by mspelman07 — CHANGES_REQUESTED** [review]
Generally looks good. Just a few doc string updates.

File: improver/psychrometric_calculations/psychrometric_calculations.py [hunk 5]
```diff
@@ -890,6 +896,17 @@ def process(self, cubes: Union[CubeList, List[Cube]]) -> Cube:
         wet_bulb_temperature.convert_units("celsius")
         wet_bulb_integral.convert_units("K m")
 
+        if self.model_id_attr:
+            if (
+                wet_bulb_temperature.attributes[self.model_id_attr]
+                != wet_bulb_integral.attributes[self.model_id_attr]
+            ):
+                raise ValueError(
```
[the remark this answers is not in the document]
Comment (gavinevans) [reply improver/psychrometric_calculations/psychrometric_calculations.py:908]: Added.

File: improver_tests/acceptance/test_phase_change_level.py [hunk 6]
```diff
@@ -41,6 +41,7 @@
 run_cli = acc.run_cli(CLI)
 
 
+@pytest.mark.parametrize("model_id_attr", (True, False))
```
[the remark this answers is not in the document]
Comment (gavinevans) [reply improver_tests/acceptance/test_phase_change_level.py:44]: I've updated the docstring.

File: improver_tests/acceptance/test_wet_bulb_temperature.py [hunk 7]
```diff
@@ -41,16 +41,22 @@
 run_cli = acc.run_cli(CLI)
 
 
-def test_basic(tmp_path):
+@pytest.mark.parametrize("model_id_attr", (True, False))
+def test_basic(tmp_path, model_id_attr):
```
[the remark this answers is not in the document]
Comment (gavinevans) [reply improver_tests/acceptance/test_wet_bulb_temperature.py:45]: Updated.

File: improver_tests/acceptance/test_wet_bulb_temperature_integral.py [hunk 8]
```diff
@@ -41,13 +41,19 @@
 run_cli = acc.run_cli(CLI)
 
 
-def test_basic(tmp_path):
+@pytest.mark.parametrize("model_id_attr", (True, False))
+def test_basic(tmp_path, model_id_attr):
```
[the remark this answers is not in the document]
Comment (gavinevans) [reply improver_tests/acceptance/test_wet_bulb_temperature_integral.py:45]: Updated.

**Review by mspelman07 — APPROVED** [review]

**Pull request merged**